packages/marshal/src/marshal.js

```diff
@@ -333,11 +345,22 @@ export const makeMarshal = (
      * representation. Rather, fullRevive must validate that.
      *
      * @param {Encoding} rawTree must be hardened
+     * @param {string} key
      */
-    function fullRevive(rawTree) {
+    function fullRevive(rawTree, key) {
```

Comment:
If we do swap the order of params, we should probably pick a name other than `fullRevive`. We probably should anyway.